# HG changeset patch
# Parent  4364632d507030b0c764bc9626820d91a23b4096
Add all waveform types in lightsc.py

diff --git a/examples/lightsc.py b/examples/lightsc.py
--- a/examples/lightsc.py
+++ b/examples/lightsc.py
@@ -60,6 +60,38 @@
     )
 
 
+def half_sine(socket, target, h, s, b, k, period, cycles, transient=True):
+    return set_waveform(
+        socket, target, "HALF_SINE", h, s, b, k,
+        cycles=cycles,
+        period=period,
+        skew_ratio=0.5,
+        transient=transient
+    )
+
+
+def triangle(socket, target, h, s, b, k,
+             period, cycles, peak=0.5, transient=True):
+    return set_waveform(
+        socket, target, "TRIANGLE", h, s, b, k,
+        cycles=cycles,
+        period=period,
+        skew_ratio=peak,
+        transient=transient
+    )
+
+
+def pulse(socket, target, h, s, b, k, period, cycles,
+          duty_cycle=0.5, transient=True):
+    return set_waveform(
+        socket, target, "PULSE", h, s, b, k,
+        cycles=cycles,
+        period=period,
+        skew_ratio=duty_cycle,
+        transient=transient
+    )
+
+
 def power_on(socket, target):
     return jsonrpc_call(socket, "power_on", {"target": target})
 
